src/counter.rs

```diff
@@ -0,0 +1,69 @@
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::time::{Duration, UNIX_EPOCH, SystemTime};
+
+pub struct Counter {
+    pub name: &'static str,
+    pub counts: AtomicUsize,
+    pub nanos: AtomicUsize,
+    pub times: AtomicUsize,
+    pub lograte: usize,
+}
+
+macro_rules! STATIC_COUNTER {
```

Comment:
Can you make lowercase? And is there a reason for it to include the word `static`?

Reply:
because its for static initialization of the counter.  its kind of a special thing, we are instrumenting the code with a global